### PR TITLE
More robust getting web3.

### PR DESCRIPTION
**Pull request by juharris**
I found that with the current setup web3 wasn't being found in some cases with my app even though this one works fine. I think it's because of the mentioned race condition and the event listener for `"load"` wasn't getting triggered.

Sorry Node/JS isn't my main language please let me know if there are style issues and cases I didn't catch.

### CODE REVIEW

File: client/src/utils/getWeb3.js [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
+    }
+    // Legacy dapp browsers...
+    else if (window.web3) {
+      // Use Mist/MetaMask's provider.
+      const web3 = window.web3;
+      console.log("Injected web3 detected.");
+      resolve(web3);
+    }
+    // Fallback to localhost; use dev console port by default...
+    else {
```
Comment (juharris) [inline client/src/utils/getWeb3.js]: Actually maybe this `else` case should only be done if the "load" event is being triggered?

**Comment by adrianmcli** [comment]
Hi @juharris thanks for the PR. Do you mind detailing to me, in an issue, what exactly is the problem? I would like to make sure I can replicate the bug first before incorporating any changes.

**Comment by juharris** [comment]
I think it was because I had a bunch of setup logic in my constructor or stuff that needed to run before `getWeb3` was called. I probably could refactor but this won't be obvious to everyone so I thought that I should make `getWeb3` more robust to handle cases where the event listener for "load" would already be triggered.

**Comment by adrianmcli** [comment]
I think this is a little out of scope for this project, but we are actually working on this in an upcoming tool. We take the path of checking `document.readyState` instead. You can have a look [here](https://github.com/trufflesuite/drizzle-utils/blob/master/packages/get-web3/index.js#L37-L40).

I thank you and appreciate the effort, but I think I will be closing this for now!

**Comment by juharris** [comment]
Ah that solution looks much better.  Thanks for reviewing!

**Pull request closed without merging**